### PR TITLE
new skaffold site

### LINKED ISSUE #227: Provide getting started for windows

### LINKED ISSUE #498: add docs on the environment variables/expressions used
We've a few special environment variables/expressions used right now in the `envTemplates` expressions (and maybe soon on the same for helm charts too https://github.com/GoogleContainerTools/skaffold/issues/486) plus this PR: https://github.com/GoogleContainerTools/skaffold/pull/495. 

So we should write a little doc that describes all the different values available in the different templates

### LINKED ISSUE #519: Create a website for Skaffold
Let's start small with maybe Github Pages and a custom domain

### LINKED ISSUE #628: Helm deployment docs
Helm deployment docs need some love, as currently is not really clear where is your code and how it is related with chart releases

### LINKED ISSUE #629: Revamp documentation
The docs need a bit more love. 
This is an overarching issue that would be done when a more extensive documentation is done. 

- [x] Get to an agreement on tooling 
- [x] Get agreed on an outline 
- [x] document all the outstanding [docs issues](https://github.com/GoogleContainerTools/skaffold/issues?q=is%3Aissue+is%3Aopen+label%3Adocumentation)
- [x] implement all the individual parts of the outline 

The main goal is to get to a place where all features and knobs are well described, after which, we can enforce a contributor policy that adding a contribution would only considered done if there is a corresponding change in documentation and the annotated skaffold.yaml.

### LINKED ISSUE #1076: Add new features to README
**Automatic port-forwarding**
Skaffold automatically port-forwards any container ports in pods.

**File syncing**
Skaffold can automatically sync files to a running container and skip a rebuild. Runs your pods 

**skaffold init**
Skaffold can generate its own skaffold configuration file for "source-to-deployment"

**deployment labeling**

Probably some others I missed. We should incorporate these into our new site/README

### LINKED ISSUE #1327: Define default-repo in the concepts docs
This would be beneficial both currently + in the new site.

### PR DESCRIPTION
**Pull request by balopat**
Skaffold site is living currently on the `skaffold_site` feature branch for collaboration. 

This PR is tracking the latest version of the docs and makes sure there are no merge conflicts with master. 

To contribute changes to the site, raise a PR against the `skaffold_site` base, then after the change is merged, add the `docs-modification` label to this PR and our bot should deploy a preview version that will be accessible through an ephemeral URL. 

---------------------

Fixes #629.
Fixes #628.
Fixes #498.
Fixes #227.
Fixes #1076 .
Fixes #519.
Fixes #1327.

TODO:

[X] a release script
[X] a PR review workflow (netlify or something similar)
[X ] a front page (and a new theme)

### CODE REVIEW

**Comment by dgageot** [comment]
@balopat Can it be merged? I'd find it much easier to contribute fixes an improvements.

File: docs/content/en/docs/getting-started/_index.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+</ol>
+
+{{< alert title="Note" >}}
+
+If you use a non-local solution, your Docker client needs to be configured
+to push Docker images to an external Docker image registry. For setting up
+Docker with Google Container Registry, see <a href=https://cloud.google.com/container-registry/docs/quickstart>Google Container Registry Quickstart</a>.
+{{< /alert >}}
+
+## Installing Skaffold
```
Comment (dgageot) [inline docs/content/en/docs/getting-started/_index.md]: This section is empty on the website: http://35.236.7.76:1313/docs/getting-started/#installing-skaffold

File: docs/content/en/_index.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+
+{{< blocks/section >}}
+{{% blocks/feature icon="fa-download" title="Install Skaffold" %}}
+[Get Skaffold on Linux, macOS, and Windows.](/docs/getting-started/#installing-skaffold)
+{{% /blocks/feature %}}
+
+
+{{% blocks/feature icon="fab fa-github" title="Contributions welcome!" %}}
+We'd love to accept your patches and contributions to this project. See the [Contribution Guide](https://github.com/GoogleContainerTools/skaffold/blob/master/CONTRIBUTING.md) to get started.
```
Comment (dgageot) [inline docs/content/en/_index.html]: `We’d love to accept your contributions`?

File: docs/content/en/docs/_index.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+menu:
+  main:
+    weight: 20
+---
+
+Skaffold is a command line tool that facilitates continuous development for
+Kubernetes applications. You can iterate on your application source code
+locally then deploy to local or remote Kubernetes clusters. Skaffold handles
+the workflow for building, pushing and deploying your application. It can also
+provide building blocks and describe customizations for a CI/CD pipeline.
```
Comment (dgageot) [inline docs/content/en/docs/_index.md]: `It also provides?`

File: docs/content/en/docs/how-tos/deployers/_index.md [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+When skaffold deploys an application the following steps happen: 
+* the skaffold deployer _renders_ the final kubernetes manifests: skaffold replaces the image names in the kubernetes manifests with the final tagged image names. 
+Also, in case of the more complicated deployers the rendering step involves expanding templates (in case of helm) or calculating overlays (in case of kustomize). 
+* the skaffold deployer _deploys_ the final kubernetes manifests to the cluster
+
+At this moment, Skaffold supports the following tools for deploying applications:
+
+* [`kubectl`](#deploying-with-kubectl) 
+* [Helm](#deploying-with-helm) 
+* [Kustomize](#deploying-with-kustomize)
```
Comment (dgageot) [inline docs/content/en/docs/how-tos/deployers/_index.md]: lowercase kustomize

File: docs/content/en/docs/getting-started/_index.md (hunk 1 above)
Comment (balopat) [reply docs/content/en/docs/getting-started/_index.md]: hmmm...tabs are acting up.

**Comment by balopat** [comment]
@dgageot PTAL I'd be okay with merging now even though it's far from perfect, but it's better than what we have, yay :)

**Comment by dgageot** [comment]
🎉 💃

**Comment by hypnoglow** [comment]
Hi @balopat

Any reason `Gopkg.lock` was reverted from dep 0.5.* format to dep 0.4.* format?

**Pull request merged**